src/Autofac.Integration.Mef/RegistrationExtensions.cs

```diff
@@ -423,10 +427,136 @@ private static void ProcessExportDefinition(ContainerBuilder builder, Func<Expor
             }
         }
 
+        private static Func<object, LazyMemberInfo> getReflectionMemberImportDefinition;
+        private static Func<object, Lazy<ParameterInfo>> getReflectionParameterImportDefinition;
+
+        private static bool TryGetLazyType(this ContractBasedImportDefinition definition, out Type resultType, out Type lazyType)
+        {
+            // There are a couple of classes that are internal that provide us some information we can use to
+            // properlay guage if we need to do our lazy activation or not.
+            var definitionType = definition.GetType();
+            LazyMemberInfo? lazyMemberInfo = null;
+            resultType = null;
+            switch (definitionType.Name)
+            {
+                case "ReflectionMemberImportDefinition":
+                case "PartCreatorMemberImportDefinition":
+                    if (getReflectionMemberImportDefinition == null)
+                    {
+                        var def = Expression.Parameter(typeof(object), nameof(definition));
```

Comment:
The PartCreator type extended the Reflection type, but you did catch a bug where it could compile against the PartCreator type which would cause a subsequent call to fail.  I'm not too sure of the likelihood where we could get both instances to come through, but it has been resolved with the updated building logic.

Also added comment to explain why they leverage the same function.

Reply:
No longer needed, due to use of methods available on ReflectionModelServices